test/e2e/ps_test.go

```diff
@@ -425,6 +425,31 @@ var _ = Describe("Podman ps", func() {
 		Expect(session.OutputToString()).To(ContainSubstring("0.0.0.0:2000-2006"))
 	})
 
+	It("podman ps test with multiple port range", func() {
+		session := podmanTest.RunTopContainer("")
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		session = podmanTest.Podman([]string{
+			"run", "-dt",
+			"-p", "3000-3001:3000-3001",
+			"-p", "3100-3102:4000-4002",
+			"-p", "30080:30080",
+			"-p", "30443:30443",
+			"-p", "6000-6066",
+			"-p", "8000:8080",
+			ALPINE, "top"},
+		)
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		session = podmanTest.Podman([]string{"ps", "--format", "{{.Ports}}"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.OutputToString()).To(ContainSubstring(
+			"0.0.0.0:3000-3001->3000-3001/tcp, 0.0.0.0:3100-3102->4000-4002/tcp, 0.0.0.0:6000-6066->6000-6066/tcp, 0.0.0.0:8000->8080/tcp, 0.0.0.0:30080->30080/tcp, 0.0.0.0:30443->30443/tcp",
+		))
```

Comment:
This seems slightly fragile, as it seems conceivable the sorting could change over time.  For example, sorting ranges before single-port mappings could flip.  While meaningless to humans, the test maybe unnecessarily sensitive to the order.  Is the cost of output-checking each item separately, worth the benefit of a less pedantic (but still effective) test?

Reply:
My idea was to test the sorting as well because we explicitly sort it in the code. The fix mainly targets for use cases where people add multiple port ranges to a single Podman command, which is right now broken. Not sure how to improve the test to be easier to reflect. Do you have any idea?

Reply:
Oh okay, that's fine if the intention is to also verify sorting.  In that case I would as you to consider putting that in a comment?  That would be really helpful to some future developer who manages to break both sorting and fail the test  :grin: